nflows/transforms/standard.py

```diff
@@ -1,82 +1,92 @@
 """Implementations of some standard transforms."""
 
+from typing import Optional
+import warnings
+import numpy as np
 import torch
-
+from nflows.utils.torchutils import ensure_tensor
 from nflows.transforms.base import Transform
 
 
 class IdentityTransform(Transform):
     """Transform that leaves input unchanged."""
 
-    def forward(self, inputs, context=None):
-        batch_size = inputs.shape[0]
-        logabsdet = torch.zeros(batch_size)
+    def forward(self, inputs: torch.Tensor, context=Optional[torch.Tensor]):
+        num_batches = inputs.shape[0]
+        logabsdet = torch.zeros(num_batches)
         return inputs, logabsdet
 
-    def inverse(self, inputs, context=None):
+    def inverse(self, inputs: torch.Tensor, context=Optional[torch.Tensor]):
         return self(inputs, context)
 
 
-class AffineScalarTransform(Transform):
-    """Computes X = X * scale + shift, where scale and shift are scalars, and scale is non-zero."""
-
-    def __init__(self, shift=None, scale=None):
+class PointwiseAffineTransform(Transform):
+    def __init__(
+        self,
+        shift: torch.Tensor = torch.tensor(0.0),
+        scale: torch.Tensor = torch.tensor(1.0),
+    ):
         super().__init__()
+        shift, scale = map(ensure_tensor, (shift, scale))
 
-        if shift is None and scale is None:
-            raise ValueError("At least one of scale and shift must be provided.")
-        if scale == 0.0:
-            raise ValueError("Scale cannot be zero.")
+        # reject scales < ~0 (upto dtype precision)
+        is_scale_positive = torch.isfinite(torch.log(scale)).any()
```

Comment:
Why not a much simpler `(scale <= 0).any()`?

Reply:
That's right! This is a mental leftover from `scale` potentially being a scalar. We had then:  
> In [15]: (1E-46>0, tensor(1E-46)>0)                                                                 
Out[15]: (True, tensor(False))

and I wanted to test closest to the point of (potential) breakage - conservatively. But with `scale` a tensor, this no longer makes sense.